### PR TITLE
Module 7

### PR DESCRIPTION
**Pull request by A-J1993**

### CODE REVIEW

File: test_e2e/test_app_selenium.py [hunk 1]
```diff
@@ -56,7 +56,13 @@ def test_app():
 
 @pytest.fixture(scope = "module")
 def driver():
-    with webdriver.Firefox() as driver:
+    #with webdriver.Firefox() as driver:
+    #    yield driver
```
Comment (JackMeadDev) [inline test_e2e/test_app_selenium.py]: Let's delete rather than commenting, we can always recover it with git

Comment (JackMeadDev) [inline test_e2e/test_app_selenium.py]: Let's delete rather than commenting, we can always recover it with git

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  - docker run my-test-image tests
+  - docker run --env SECRET_KEY --env TRELLO_TODOID --env TRELLO_DONEID --env TRELLO_BOARDID --env TRELLO_KEY --env TRELLO_TOKEN my-test-image test_e2e
+
+env:
+  global:
+    - SECRET_KEY=secret-key
+    - TRELLO_TODOID=5fb51cfd98afad055cb08c82
+    - TRELLO_DONEID=5fb51cfd98afad055cb08c84
+    - TRELLO_BOARDID=5fb51cfd98afad055cb08c81
+    - secure: "wV+sfKT3Nsz2ll3GKf8SDwrbwTHvgTbz/9OHcQalaxW6tOS2LdX1GcHifTlK1JgNnEcORkvjk9asjcw3B1H9NbSZdw0+8pQn5//she0bALHIYDCYcZuRHqwjHtw3Y41YBbFE/8God/e6U/IwYy5Hrlt/SZavCmK/aw8ivfwehiZAe8oPhDq6KnJSErMgmdd2gS+xvClQuhK0p4isRel9XXIuARm1m8viAZtixQdbWH8lZ2pNEzQNdsLmJJ2WazjqkYz34g/sXYJcs8GIk5PlPj3HUWtRbX3DuikzmQdReuxkD7Q/3ONymNDb9xXd7jZGgCeMVGHd+A7Tu8B7eT9Vi7sUm9egs3NPTDD0tFoiKC8DR++WKMxuE8VaEMOnI/CfZaxOnqJOshCViCtOhxV6Il2KGtNjqjOZuf1IlUynD3yvI7V0trzyxwy728H8lCVSIfv7+dZ2Z5k3GX258ny6C00iD3zrqJ/OTv4AYPK3YEiyW/nRFhuDzrPI5JozQM8ibcvCrbYGf7qi6UWZxZ5X2BT1+KIk2ZVRUnSYJRbLeIsmy2MT2wiAcdx4itlrPwWmJiBRN4U8nhqjz/AUsWe+0cHzR1CAfsyjMeQOIEm70HAsU/dFaazHHCgaGFGy/2LyjH/de4SxjSvnF10cvH/7baOMD9aRSY4YeMIs3yNVLdE="
```
Comment (JackMeadDev) [inline .travis.yml]: Worth mentioning that these secure variables can be hard to track later - the order here should match the order that Travis spits them out at the top of its output

File: Dockerfile [hunk 3]
```diff
@@ -14,4 +14,28 @@ ENTRYPOINT poetry run gunicorn --workers=2 "todo_app.app:create_app()" --bind 0.
 
 FROM base as development
 RUN poetry install
-ENTRYPOINT poetry run flask run --host=0.0.0.0
\ No newline at end of file
+ENTRYPOINT poetry run flask run --host=0.0.0.0
+
+FROM base as test
+RUN poetry install
+RUN poetry add selenium
```
Comment (JackMeadDev) [inline Dockerfile]: This line is unnecessary - you're already tracking selenium in your pyproject.toml

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+ branches:
+  only:
+    - module-7
+
+services:
+  - docker
+
+script:
+  - echo "hello world"
```
Comment (JackMeadDev) [inline .travis.yml]: We can remove this line now

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+ branches:
+  only:
+    - module-7
```
Comment (JackMeadDev) [inline .travis.yml]: Don't forget to remove this when you're ready to merge; we want these tests to run on every branch going forward

**Review by JackMeadDev — CHANGES_REQUESTED** [review]
This looks good, I can see your tests are running & passing in Travis and I was able to run your tests locally - I've left a couple of thoughts below but the main issue is a small one; we should update the Readme to mention that running the tests in docker is an option, and document how those can be ran

File: Dockerfile (hunk 3 above)
Comment (A-J1993) [reply Dockerfile]: I'm thinking of keeping it since I'm scared of breaking something

File: .travis.yml (hunk 5 above)
Comment (A-J1993) [reply .travis.yml]: Done

File: Dockerfile (hunk 3 above)
Comment (JackMeadDev) [reply Dockerfile]: A bad practice to get into! Now that you have this pipeline you can safely change it - if it causes issues then your pipeline will fail & tell you, and worst-comes-to-worst you can just add it back 😄

File: .travis.yml [hunk 6]
```diff
@@ -1,21 +1,17 @@
...
 script:
-  - echo "hello world"
   - docker build --target test --tag my-test-image . 
   - docker run my-test-image tests
   - docker run --env SECRET_KEY --env TRELLO_TODOID --env TRELLO_DONEID --env TRELLO_BOARDID --env TRELLO_KEY --env TRELLO_TOKEN my-test-image test_e2e
 
 env:
   global:
     - SECRET_KEY=secret-key
+    - secure: "wV+sfKT3Nsz2ll3GKf8SDwrbwTHvgTbz/9OHcQalaxW6tOS2LdX1GcHifTlK1JgNnEcORkvjk9asjcw3B1H9NbSZdw0+8pQn5//she0bALHIYDCYcZuRHqwjHtw3Y41YBbFE/8God/e6U/IwYy5Hrlt/SZavCmK/aw8ivfwehiZAe8oPhDq6KnJSErMgmdd2gS+xvClQuhK0p4isRel9XXIuARm1m8viAZtixQdbWH8lZ2pNEzQNdsLmJJ2WazjqkYz34g/sXYJcs8GIk5PlPj3HUWtRbX3DuikzmQdReuxkD7Q/3ONymNDb9xXd7jZGgCeMVGHd+A7Tu8B7eT9Vi7sUm9egs3NPTDD0tFoiKC8DR++WKMxuE8VaEMOnI/CfZaxOnqJOshCViCtOhxV6Il2KGtNjqjOZuf1IlUynD3yvI7V0trzyxwy728H8lCVSIfv7+dZ2Z5k3GX258ny6C00iD3zrqJ/OTv4AYPK3YEiyW/nRFhuDzrPI5JozQM8ibcvCrbYGf7qi6UWZxZ5X2BT1+KIk2ZVRUnSYJRbLeIsmy2MT2wiAcdx4itlrPwWmJiBRN4U8nhqjz/AUsWe+0cHzR1CAfsyjMeQOIEm70HAsU/dFaazHHCgaGFGy/2LyjH/de4SxjSvnF10cvH/7baOMD9aRSY4YeMIs3yNVLdE="
```
Comment (JackMeadDev) [inline .travis.yml:12]: Possibly unclear on what I said before, but no harm - I was just advising that if you ever need to figure out what variable this represents in future you can refer to the order Travis spits them out, not that there's a particular order in which you should list them

**Review by JackMeadDev — APPROVED** [review]
Thanks for updating those things, I'm happy for you to merge when you're ready 😄

**Pull request merged**